libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,27 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
+        if (!baro->all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (checks_to_perform & ARMING_CHECK_GPS) {
+            const AP_GPS *gps = AP_GPS::get_singleton();
+            if (gps->status() >= AP_GPS::GPS_OK_FIX_3D) {
+                if (gps->get_vdop() < 170 && gps->num_sats() > 8) {
+                    float baro_altasl = baro->get_altitude_difference(101325.0f, baro->get_pressure());
+                    float gps_alt = gps->location().alt*0.01f;
+                    if (baro_altasl > (gps_alt+100.0f) || baro_altasl < (gps_alt-100.0f)) {
+                        check_failed(ARMING_CHECK_BARO, report, "Barometer alt error");
+                        return false;
+                    }
+                }
+            } else {
+                check_failed(ARMING_CHECK_BARO, report, "Barometer alt waiting for GPS alt fix");
```

Comment:
You mean vdop 170 and 8 sats are not enough?
What you think else can be done here?

For other sensors, like Dual GPS they are checked for position variance before arming.
Also GPS speed is checked against IMU readings until speed error is less than some reasonable value.

So I think it's consistent with such behavior.

Reply:
No, that's not what I meant. If you don't have 3D Fix, you can't arm because the check fails. If you have 3D Fix but you don't have 8 sats or VDOP is higher than 170 then no check is performed and you can arm.

Reply:
Fixed. Thanks

Reply:
I've increased the vdop to 201 to be compatible with SITL vdop.